src/lib/os.pl

```diff
@@ -137,9 +137,8 @@
 argv(Argv) :-
     can_be(list, Argv),
     '$argv'(Argv0),
-    ( member("--", Argv0) ->
-        once(append(Argv1, ["--"|Argv], Argv0)),
-        \+ member("--", Argv1)
+    ( append(_, ["--"|Argv1], Argv0) ->
+        Argv = Argv1
```

Comment:
I think this is `( append(_, ["--"|Argv], Argv0) -> true` ?

Reply:
That would be the orginal code prior to this PR.
Due the order in which the sub-goals are evaluated they are not equivalent.

By adding `Argv1` we prevent `Argv` from influencing the solution (if any) `append/3` finds first.
Without `Argv1` `append/3` may find a later suffix if the argument list contains multiple `--` and `Argv` doesn't match at the first `--`, resulting in an incorrect solution.

Should the evaluation behaviour change in the futur (i.e. the `=/2` here is inlined) then the added test should detect this.

Reply:
Perfect, thank you a lot!